internal/gengapic/genrest.go

```diff
@@ -603,8 +603,22 @@ func (g *generator) pagingRESTCall(servName string, m *descriptor.MethodDescript
 	p("")
 	p("  unm.Unmarshal(buf, resp)")
 	p("  it.Response = resp")
-	// TODO(dovs): handle map pages
-	p("  return resp.Get%s(), resp.GetNextPageToken(), nil", snakeToCamel(elemField.GetName()))
+	repeatedField, elts := fmt.Sprintf("resp.Get%s()", snakeToCamel(elemField.GetName())), ""
+	if pt.mapValueTypeName != "" {
+		elts = "elts"
+		p("")
+		p("    elts := make([]%s, 0, len(%s))", pt.elemTypeName, repeatedField)
```

Comment:
This will work when the map's value type is a list (ie `repeated`)? That's the GCE use case: each page is a map of strings to list, and each item being iterated should be a (string, List) pair. (The code I have pending for Showcase to test mapped responses implements this. I can put out a draft PR so you can test against it.)

Reply:
I think what this code is doing is actually building a slice of the the key-value pair structs from the map. Basically "flattening" the `map` into a list.

The value in the map will be a message that has a slice inside of it, if I'm not mistaken.

Reply:
Noah is correct. This code is reconstructing the proto-implementation of having repeated key/value pairs.
If we have the following proto message:
```proto
message List {
    repeated string items = 1;
}

message Request {
    string next_page_token = 1;
    map<string, List> lists = 2;
}
```

Gives us the following helper structs:
```go
type ListPair struct {
    Key string
    Value *List
} 
```

The `elts` slice is then of type `[]ListPair`.

Reply:
OK, what you describe is correct behavior. I just wanted to make sure it works when the map values are lists, but as you reminded me. `repeated`s need to be embedded in `message`s, so my concern is the same as ensuring that this works when the map values are messages instead of primitives. In your companion PR, the return proto map has primitive value type, so just wanting to make sure this works with messages....

Reply:
Noah's comment about full message type names prompted me to add a test where the value type is a message, which fails in an interesting and unexpected but not serious way. It is mostly fine.